syntax = "proto3";
package feast.core;

option go_package = "github.com/feast-dev/feast/go/protos/feast/core";
option java_outer_classname = "VectorFeatureViewProto";
option java_package = "feast.proto.core";

import "google/protobuf/duration.proto";
import "google/protobuf/timestamp.proto";
import "feast/core/Entity.proto";
import "feast/core/FeatureView.proto";

// Next available id: 5
message VectorFeatureView {
    // using composition to include inherited feature view
    FeatureView feature_view = 1;

    // FeatureView doesn't preserve entities when converting to protos hence storing them separately here to work
    // around this issue
    repeated Entity entities = 2;

    // name of the vector field
    string vector_field = 3;

    // dimensions of the vectors
    int32 dimensions = 4;

    // the index algorithm to be used
    string index_type = 5;
}
